lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/AzureStorageClient.java

```diff
@@ -172,28 +183,26 @@ private Document blobItemToDoc(BlobItem blob, InputStream is, String decompresse
     }
 
     // unable to get the decompressed size via inputStream
-    if (getFileContent) {
+    if (params.shouldGetFileContent()) {
       doc.setField(FileConnector.CONTENT, is.readAllBytes());
     }
 
     return doc;
   }
 
-  private boolean isValid(BlobItem blob) {
+  private boolean isValid(BlobItem blob, TraversalParams params) {
     if (blob.isPrefix()) return false;
 
-    return shouldIncludeFile(blob.getName(), includes, excludes);
+    return params.shouldIncludeFile(blob.getName());
   }
 
-  public static void validateOptions(Map<String, Object> cloudOptions) {
-    if (!cloudOptions.containsKey(AZURE_CONNECTION_STRING) &&
-        !(cloudOptions.containsKey(AZURE_ACCOUNT_NAME) && cloudOptions.containsKey(AZURE_ACCOUNT_KEY))) {
-      throw new IllegalArgumentException("Either '" + AZURE_CONNECTION_STRING + "' or '" + AZURE_ACCOUNT_NAME + "' & '" + AZURE_ACCOUNT_KEY + "' has to be in cloudOptions for AzureStorageClient.");
-    }
+  public static boolean validOptions(Config cloudOptions) {
```

Comment:
this method seems to only be called from within validateOptions()
please move this logic inside validateOptions and eliminate this static validOptions method